packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -591,6 +594,24 @@ abstract class FlutterCommand extends Command<void> {
       valueHelp: 'foo=bar',
       splitCommas: false,
     );
+    if (enableDefineConfigJsonFileOption) {
+      useDefineConfigJsonFileOption();
+    }
+  }
+
+  void useDefineConfigJsonFileOption() {
+    argParser.addOption(
+      FlutterOptions.kDartDefineFromFileOption,
+      help: 'The path of a json format file where flutter define a global constant pool. '
+          'Json entry will be available as constants from the String.fromEnvironment, bool.fromEnvironment, '
+          'int.fromEnvironment, and double.fromEnvironment constructors;the key is json filed key, the value is json value.',
+      valueHelp: 'use-define-config.json'
+    );
+    argParser.addFlag(
```

Comment:
Is this feature generally useful to Flutter developers?

Reply:
Useful, because under our internationalization, we need to configure different package names, different configuration files, and different third-party accounts. If the traditional mode is used, the command line will be too long, which is not conducive to maintenance.

If changed to this
1. It can facilitate centralized configuration, separate in debug and release modes, which is conducive to configuration management and different package management, and achieves good decoupling
2. You can operate project variables in android grade, and you can directly use variables in plist under the apple platform, which is very convenient for parameter changes

Because our company is mainly facing overseas and needs to be localized, the promotion and verification are very efficient and convenient at present. The development only needs to maintain a set of config.json in test mode, and then use jenkins to type out Brazil, Southeast Asia, and Europe. Package [We need to sub-contract operation], currently ranked in Brazil, flutter can greatly reduce our development time, and it is very stable, and other android and iOS personnel can also start very quickly. 👍🏻

Reply:
> kEnableDartDefineFromFileRawValue

The usage rate of this is indeed not as wide as the previous flag usage scenarios. This is just to facilitate the enumeration of how many variables currently exist.
1. It can map out global beans for centralized management
2. It is convenient to enumerate existing variables. Now there is no way to enumerate existing variables.

At present, it is only used in our project. If everyone thinks it is really unnecessary, we can remove it.

Reply:
sounds good to me!